dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/PickleUtils.cs

```diff
@@ -182,9 +233,8 @@ public static int bytes_to_integer(byte[] bytes, int offset, int size)
 			throw new PickleException("decoding float: too few bytes");
 	    }
     	if(BitConverter.IsLittleEndian) {
-			// reverse the bytes to make them littleendian for the bitconverter
-			byte[] littleendian={ bytes[3+offset], bytes[2+offset], bytes[1+offset], bytes[0+offset] };
-			return BitConverter.ToSingle(littleendian,0);
+            int intVal = BinaryPrimitives.ReadInt32BigEndian(bytes.AsSpan(offset));
```

Comment:
according to msdn this is a NetStandard 2.1 level method. Currently, the project is targeting NetStandard 2.0. Is there an alternative?

Reply:
also, shouldn't it be ReadInt32LittleEndian() here

Reply:
> according to msdn this is a NetStandard 2.1 level method. Currently, the project is targeting NetStandard 2.0. Is there an alternative?

The docs are misleading, unfortunately.  This type is built into .NET Core 2.1 and is part of .NET Standard 2.1, but the NuGet package has multiple targets, all the way down to .NET Standard 1.1.  When targeting a platform that has the support built-in, it just type forwards to those built-in implementations.  When targeting a platform that doesn't, the .dll includes the necessary types.  Which is why it successfully builds and passes CI even though the library targets .NET Standard 2.0 :-)

Reply:
> also, shouldn't it be ReadInt32LittleEndian() here

I don't think so.  If we're here, BitConverter.IsLittleEndian is true, which means ToSingle expects little endian, but the code was explicitly reversing the bytes, which means they were big endian... so the replacement for the byte[] allocation with the reversed bytes and a subsequent call to ToSingle is just to ReadInt32BigEndian.

Have a I misunderstood?

Reply:
you're correct